src/component/DefaultLegendContent.tsx

```diff
@@ -170,9 +171,14 @@ export class DefaultLegendContent extends PureComponent<Props> {
         return null;
       }
 
-      const entryValue = _.isFunction(entry.value) ? entry.value() : entry.value;
-      const color = entry.inactive ? inactiveColor : entry.color;
+      // Do not render entry.value as functions. Always require static string properties.
+      const entryValue = !_.isFunction(entry.value) ? entry.value : null;
+      warn(
```

Comment:
Do you think a warning or an invariant would be better in this case?

Warnings are still logged in production and aren't in your face enough. 

Invariant will cause an error only in development (in your face) and will be ignored in production builds. 

Mostly curious on thought of what is the better UX

Reply:
I like throwing an error better actually as well for the UX.

Side note, maybe I'm misunderstanding our `warn()` util, but I assumed that's what the `isDev` check was for here (but as a warning instead of an error):
https://github.com/recharts/recharts/blob/21ce248c06ab75893f17b6308deedbdb7a67238e/src/util/LogUtils.ts#L2-L5

I'll switch to using the tiny-invariant dependency instead.

Reply:
Ah yeah, you're not misunderstanding - I just don't have all of the context either since these things existed long before me 😅. Sounds good though

Reply:
@ckifer Makes sense. 😅
One quick thing I just thought about was backwards compatibility. Throwing an error here would force all existing projects that currently have functions as dataKeys but aren't including the name in a `<Legend/>` to have fatal errors in dev environments. Those scenarios admittedly would likely be a result of dead code. But do we think that very specific, potentially semi-breaking change is worth opting for an error over a warning here?

Reply:
oof, you're right.

Better to have a warning from recharts than the warning they get now from react (functions not valid as a child). 

Next MV this should be changed to invariant, lets keep it a warning for now

Reply:
Cool, sounds good!

Reply:
This lib has had a lot of unfortunate breaking changes in minor/patch version releases throughout its life. If possible, I'd like to prevent that moving forwards so we'll keep it as a warning in this iteration.